Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.py

```diff
@@ -0,0 +1,1082 @@
+        publish=args.get("publish", None),
+        defaultWatchers=args.get("default_watchers", None),
+    )
+
+    return CommandResults(
+        readable_output=pretty_print_case_metadata(result),
+        outputs_prefix="Argus.Case",
+        outputs=result,
+        raw_response=result,
+    )
+
+
+def delete_case_command(args: Dict[str, Any]) -> CommandResults:
+    case_id = args.get("case_id", None)
+    if not case_id:
+        raise ValueError("case id not specified")
+
+    result = delete_case(caseID=case_id)
+
+    return CommandResults(
+        readable_output=pretty_print_case_metadata(result, "Case deleted"),
+        outputs_prefix="Argus.Case",
+        outputs=result,
+        raw_response=result,
+    )
+
+
+def delete_comment_command(args: Dict[str, Any]) -> CommandResults:
+    case_id = args.get("case_id", None)
+    comment_id = args.get("comment_id", None)
+    if not case_id:
+        raise ValueError("case id not specified")
+    if not comment_id:
+        raise ValueError("comment id not specified")
+
+    result = delete_comment(caseID=case_id, commentID=comment_id)
+
+    return CommandResults(
+        readable_output=pretty_print_comment(
+            result["data"], f"# #{case_id}: Deleted comment\n"
+        ),
+        outputs_prefix="Argus.Comment",
+        outputs=result,
+        raw_response=result,
+    )
+
+
+def download_attachment_command(args: Dict[str, Any]) -> Any:
+    case_id = args.get("case_id", None)
+    attachment_id = args.get("attachment_id", None)
+    if not case_id:
+        raise ValueError("case id not specified")
+    if not attachment_id:
+        raise ValueError("attachment id not specified")
+
+    result = download_attachment(caseID=case_id, attachmentID=attachment_id)
+
+    return fileResult(attachment_id, result.content)
+
+
+def edit_comment_command(args: Dict[str, Any]) -> CommandResults:
+    case_id = args.get("case_id", None)
+    comment_id = args.get("comment_id", None)
+    comment = args.get("comment", None)
+    if not case_id:
+        raise ValueError("case id not specified")
+    if not comment_id:
+        raise ValueError("comment id not specified")
+    if not comment:
+        raise ValueError("comment not specified")
+
+    result = edit_comment(caseID=case_id, commentID=comment_id, comment=comment)
+
+    return CommandResults(
+        readable_output=pretty_print_comment(
+            result["data"], f"# #{case_id}: Updated comment\n"
+        ),
+        outputs_prefix="Argus.Comment",
+        outputs=result,
+        raw_response=result,
+    )
+
+
+def get_attachment_command(args: Dict[str, Any]) -> CommandResults:
+    case_id = args.get("case_id", None)
+    attachment_id = args.get("attachment_id", None)
+    if not case_id:
+        raise ValueError("case id not specified")
+    if not attachment_id:
+        raise ValueError("attachment id not specified")
+
+    result = get_attachment(caseID=case_id, attachmentID=attachment_id)
```

Comment:
What does the api return in cases the attachment_id doesn't exist?

Reply:
The API raises exceptions, which are printed by return_results